src/archivematicaCommon/lib/executeOrRunSubProcess.py

```diff
@@ -145,12 +145,13 @@ def createAndRunScript(
         arguments = []
     if env_updates is None:
         env_updates = {}
-    # Output the text to a /tmp/ file
-    scriptPath = "/tmp/" + uuid.uuid4().__str__()
-    FILE = os.open(scriptPath, os.O_WRONLY | os.O_CREAT, 0o770)
+    # Output the text to a temporary file
+    tmpdir_path = os.environ.get("TMPDIR", "/tmp")
```

Comment:
What do you think of using `tempfile` here which handles the `TMPDIR`, `TEMP` and `TMP` environment variables transparently. See how [this function](https://github.com/artefactual-labs/a3m/blob/8b7a7c59f804edd6484a7c6a4619818efb1b0d66/a3m/executeOrRunSubProcess.py#L147-L176) is implemented in a3m using a context manager which deletes the file automatically at the end.

Reply:
thanks for the suggestion @replaceafill. I have updated the code to use that library